src/main/java/org/openstreetmap/atlas/geography/atlas/dynamic/DynamicAtlas.java

```diff
@@ -537,15 +537,7 @@ private boolean areaCovered(final Area area)
     private <V extends AtlasEntity> boolean entitiesCovered(final Iterable<V> entities,
             final Predicate<V> entityCoveredPredicate)
     {
-        boolean result = true;
-        for (final V item : entities)
-        {
-            if (!entityCoveredPredicate.test(item))
-            {
-                result = false;
-            }
-        }
-        return result;
+        return entities.stream().allMatch(entityCoveredPredicate::test);
```

Comment:
I think what you are looking for is this:
StreamSupport.stream(entities.spliterator(), false).allMatch(entityCoveredPredicate::test);